circle.yml

```diff
@@ -18,38 +18,28 @@ dependencies:
     - cp version.json $CIRCLE_ARTIFACTS
     - docker info
     # use circleci's docker cache workaround
+    - mkdir -p ~/cache/
     - if [ -e ~/cache/docker/image.tar ]; then echo "Loading image.tar"; docker load -i ~/cache/docker/image.tar || rm ~/cache/docker/image.tar; fi
     # build image
     - docker build -t morgoth:build .
-    # Clean up old image and save the new one
-    - mkdir -p ~/cache/docker; test '!' -e ~/cache/docker/image.tar || rm ~/cache/docker/image.tar; docker save morgoth:build > ~/cache/docker/image.tar; ls -l ~/cache/docker
+    # write the sha256 sum to an artifact to make image verification easier
+    - docker inspect -f '{{.Config.Image}}' morgoth:build | tee $CIRCLE_ARTIFACTS/docker-image-shasum256.txt
 
 test:
   pre:
     - chmod -R 777 $CIRCLE_TEST_REPORTS
   override:
-    # Run Python lint checks
-    - >
-      docker run --net host
-      -v $CIRCLE_TEST_REPORTS:/test_artifacts
-      morgoth:build flake8 --output-file /test_artifacts/flake8.txt morgoth
-    # Run JS lint checks
-    - >
-      docker run --net host
-      -v $CIRCLE_TEST_REPORTS:/test_artifacts
-      morgoth:build node_modules/.bin/eslint morgoth --format junit --output-file /test_artifacts/eslint.xml
+    # Run lint checks
+    - bin/ci/docker-run.sh therapist run --use-tracked-files
     # Run Python tests
-    - >
-      docker run --net host -e DJANGO_CONFIGURATION=Test
-      -v $CIRCLE_TEST_REPORTS:/test_artifacts
-      morgoth:build py.test --junitxml=/test_artifacts/pytest.xml
+    - bin/ci/docker-run.sh py.test --junitxml=/test_artifacts/pytest.xml morgoth/
   post:
-    # Save test artifacts for Python and Lint tests
-    - >
-      docker run -v $CIRCLE_TEST_REPORTS:/test_artifacts
-      morgoth:build flake8_junit /test_artifacts/flake8.txt /test_artifacts/flake8.xml
     # Clean up old image and save the new one
-    - mkdir -p ~/cache/docker; test '!' -e ~/cache/docker/image.tar || rm ~/cache/docker/image.tar; docker save morgoth:build > ~/cache/docker/image.tar; ls -l ~/cache/docker
+    - >
+      mkdir -p ~/cache/docker;
+      rm -f ~/cache/docker/image.tar;
+      docker save morgoth:build > ~/cache/docker/image.tar;
```

Comment:
Eep, don't do this! I had it wrong in Normandy. The cache is saved at the end of the dependency step, so moving the call to `docker save` here breaks docker caching.